sdk/nodejs/automation/cmd.ts

```diff
@@ -205,6 +207,12 @@ async function exec(
             });
         }
 
+        if (signal) {
+            signal.addEventListener("abort", () => {
+                proc.kill("SIGINT", {forceKillAfterTimeout: false});
+            });
+        }
```

Comment:
I think down below where we check `if (exitCode !== 0)` we should add `|| signal.aborted`. If the user's program _or any library code_ used by it registers a `process.on('SIGINT')`, as this integration test does:

https://github.com/pulumi/pulumi/blob/c9c30f09392d1a60c4bcf7c8c01258bba1ec5e00/sdk/go/common/util/cmdutil/testdata/graceful.js#L1-L8

Then the result will be surprising, and I don't know if this is the case today, but it seems possible to me that we race the engine and the engine considers the program "complete", i.e.: any resources in the state file not registered should be deleted.

If we're interrupted, we should _always_ return a failure status code up via gRPC so that the engine doesn't even consider deleting resources.

Reply:
May want to override the "commandResult" exitcode to a known value, as well, so that the engine doesn't see a failure with exit code 0? I'm iffy on that, would need to look at the plugin execution.

Reply:
If we sigint but the engine process finishes ok it might make sense to just return successful here? My reading of abort controller docs is it's ok for the successful case to race and win against an abort.

Reply:
> If we sigint but the engine process finishes ok it might make sense to just return successful here?

Feels like we'd be depending on an implicit and hard-to-test race. If we always error, we don't need to rely on a race to ensure that the engine won't run deletes until a program has completed successfully.

Reply:
I'm not sure I understand where the engine running deletes comes in here?  If the process has completed successfully we should report that back, no matter if someone tried to abort it.  Either way we don't have control over whether the engine finishes, that entirely depends on the timing of the user that's calling the abort.  If they try to abort the program, but it finished, and we raise a command error below that seems wrong to me, since there wasn't actually any error :thinking: